src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1497,6 +1497,56 @@ class Simplex_tree {
     return modified;
   }
 
+ public:
+  /** \brief Remove all simplices of dimension greater than a given value.
+   * @param[in] dimension Maximum dimension value.
+   * @return True if any simplex was removed, false if all simplices already had a value below the dimension.
+   * \post Note that the dimension of the simplicial complex may be lower after calling `prune_above_dimension()`
+   * than it was before. However, `upper_bound_dimension()` will return the old value, which remains a valid upper
+   * bound. If you care, you can call `dimension()` to recompute the exact dimension.
```

Comment:
This seems wrong, you modify `dimension_` which is what `upper_bound_dimension()` returns.
I think the only cases where the upper bound is not tight after this function are when it was already not tight before, and the true dimension was smaller than the argument `dimension` (in particular the function returned false). Unless I misunderstood, I would just remove this note, or rewrite it if you want to keep something there.

Reply:
Your understanding is correct. I did it in 2 rounds, and forget to remove the C++ comment. Removed on [c1f59b5](https://github.com/GUDHI/gudhi-devel/pull/837/commits/c1f59b500995373f9330b92f5e9a2e21aee5cf7c)